pkg/rancher-desktop/pages/containers/console/_id.vue

```diff
@@ -0,0 +1,781 @@
+      containerName: '',
+      containerState: '',
+      isContainerRunning: false,
+      terminal: null,
+      fitAddon: null,
+      searchAddon: null,
+      lastLogTimestamp: null,
+      pendingLogs: '',
+      isSearchExpanded: false,
+      searchTerm: '',
+    };
+  },
+  computed: {
+    ...mapGetters('k8sManager', { isK8sReady: 'isReady' }),
+    containerId() {
+      return this.$route.params.id;
+    },
+    isNerdCtl() {
+      return this.settings?.containerEngine?.name === ContainerEngine.CONTAINERD;
+    },
+    selectedNamespace() {
+      return this.settings?.containers?.namespace;
+    },
+  },
+  async mounted() {
+    this.$store.dispatch('page/setHeader', {
+      title: this.t('containers.console.title'),
+      description: '',
+    });
+
+    ipcRenderer.on('settings-read', (event, settings) => {
+      this.settings = settings;
+      this.initializeConsole();
+    });
+
+    ipcRenderer.send('settings-read');
+    this.initializeConsole();
+
+    // Add global keyboard shortcut for search
+    window.addEventListener('keydown', this.handleGlobalKeydown);
+  },
+  beforeDestroy() {
+    this.stopStreaming();
+    if (this.terminal) {
+      this.terminal.dispose();
+    }
+    ipcRenderer.removeAllListeners('settings-read');
+    window.removeEventListener('keydown', this.handleGlobalKeydown);
+  },
+  methods: {
+    async initializeConsole() {
+      if (window.ddClient && this.isK8sReady && this.settings) {
+        this.ddClient = window.ddClient;
+        await this.getContainerInfo();
+        await this.fetchLogs();
+
+        if (this.isContainerRunning) {
+          this.startStreaming();
+        }
+      }
+    },
+    async getContainerInfo() {
+      try {
+        const listOptions = { all: true };
+
+        if (this.isNerdCtl && this.selectedNamespace) {
+          listOptions.namespace = this.selectedNamespace;
+        }
+
+        const containers = await this.ddClient?.docker.listContainers(listOptions);
+
+        const container = containers.find(c => c.Id === this.containerId || c.Id.startsWith(this.containerId));
+
+        if (container) {
+          const names = Array.isArray(container.Names) ? container.Names : container.Names.split(/\s+/);
+          this.containerName = names[0]?.replace(/_[a-z0-9-]{36}_[0-9]+/, '') || container.Id.substring(0, 12);
+          this.containerState = container.State || container.Status;
+          this.isContainerRunning = container.State === 'running' || container.Status === 'Up';
+        } else {
+          this.containerName = this.containerId.substring(0, 12);
+          this.containerState = 'unknown';
+          this.isContainerRunning = false;
+        }
+      } catch (error) {
+        console.error('Error getting container info:', error);
+        this.containerName = this.containerId.substring(0, 12);
+        this.containerState = 'unknown';
+        this.isContainerRunning = false;
+      }
+    },
+    async fetchLogs(follow = false) {
```

Comment:
I don't think `follow` is ever true; if that's the case, just drop it to make the code simpler? (Though, I think the correct thing to do would be instead to always follow, and use [streaming](https://docs.docker.com/reference/api/extensions-sdk/SpawnOptions/#stream)?)